### PR TITLE
[DC-1307] Add scientific notation check to aou-ehr-file-check script

### PR DESCRIPTION
**Pull request by jp3477**
This change checks submitted integer fields for scientific notation.

### CODE REVIEW

File: omop_file_validator.py [hunk 1]
```diff
@@ -176,6 +180,32 @@ def find_blank_lines(f):
...
+
+    errors = []
+    sci_not_count = collections.defaultdict(int)
+
+    for submission_col_name in df.columns:
+        submission_column = df[submission_col_name]
+        for i, value in submission_column.items():
+            if pd.notnull(value) and re.match(SCIENTIFIC_NOTATION_REGEX,
+                                              value):
+                sci_not_count[submission_col_name] += 1
```
Comment (nishanthpp93) [inline omop_file_validator.py]: It would be helpful to catch the full list of indices here instead of just the count

**Review by nishanthpp93 — CHANGES_REQUESTED** [review]
Overall looks good. Added one suggestion

File: omop_file_validator.py (hunk 1 above)
Comment (jp3477) [reply omop_file_validator.py]: Sure, should I just put the first index of an error in case the entire column has scientific notation?

Comment (nishanthpp93) [reply omop_file_validator.py]: Yes, that should suffice. However it needs to be emphasized in the message.

Comment (jp3477) [reply omop_file_validator.py]: Sounds good, thanks.

Comment (jp3477) [reply omop_file_validator.py]: I made a change. How does it look now?

File: omop_file_validator.py [hunk 2]
```diff
@@ -176,6 +180,32 @@ def find_blank_lines(f):
...
+        for i, value in submission_column.items():
+            if pd.notnull(value) and re.match(SCIENTIFIC_NOTATION_REGEX,
+                                              value):
+                sci_not_line[submission_col_name] = i + 1
+                break
+
+    for col, linenum in sci_not_line.items():
+        e = dict(message=f"Scientific notation was found on line {linenum}. " \
+                            "Scientific notation is not allowed for integer fields.",
+                column_name=col
```
Comment (nishanthpp93) [inline omop_file_validator.py]: One small improvement would be to catch the erroneous `value` as well, maybe in a tuple
```suggestion
                sci_not_line[submission_col_name] = (value, i + 1)
                break

    for col, (value, line_num) in sci_not_line.items():
        e = dict(message=(f"Scientific notation '{value}' was found on line {line_num}. "
                      "Scientific notation is not allowed for integer fields."),
                column_name=col
```

**Review by nishanthpp93 — COMMENTED** [review]
Thanks, this looks great! Since we are only giving the first instance of the error, I added a suggestion to include the value as well to make it easier to search for.

File: omop_file_validator.py (hunk 2 above)
Comment (jp3477) [reply omop_file_validator.py]: Made another addition

**Review by nishanthpp93 — APPROVED** [review]
Looks great! Thanks for adding in the suggestions.

**Pull request merged**